diff --git a/drivers/media/usb/dvb-usb-v2/zd1301.c b/drivers/media/usb/dvb-usb-v2/zd1301.c
index 72aa6a9..5eb9182 100644
--- a/drivers/media/usb/dvb-usb-v2/zd1301.c
+++ b/drivers/media/usb/dvb-usb-v2/zd1301.c
@@ -274,11 +274,11 @@ MODULE_DEVICE_TABLE(usb, zd1301_id_table);
 static struct usb_driver zd1301_usb_driver = {
 	.name = KBUILD_MODNAME,
 	.id_table = zd1301_id_table,
-	.probe = dvb_usbv2_probe,
-	.disconnect = dvb_usbv2_disconnect,
-	.suspend = dvb_usbv2_suspend,
-	.resume = dvb_usbv2_resume,
-	.reset_resume = dvb_usbv2_reset_resume,
+	.probe = mediatree_dvb_usbv2_probe,
+	.disconnect = mediatree_dvb_usbv2_disconnect,
+	.suspend = mediatree_dvb_usbv2_suspend,
+	.resume = mediatree_dvb_usbv2_resume,
+	.reset_resume = mediatree_dvb_usbv2_reset_resume,
 	.no_dynamic_id = 1,
 	.soft_unbind = 1,
 };
