diff --git a/drivers/media/usb/dvb-usb-v2/anysee.c b/drivers/media/usb/dvb-usb-v2/anysee.c
index 7ef5a7b..4c6503f 100644
--- a/drivers/media/usb/dvb-usb-v2/anysee.c
+++ b/drivers/media/usb/dvb-usb-v2/anysee.c
@@ -48,7 +48,7 @@ static int anysee_ctrl_msg(struct dvb_usb_device *d,
 
 	/* We need receive one message more after dvb_usb_generic_rw due
 	   to weird transaction flow, which is 1 x send + 2 x receive. */
-	ret = dvb_usbv2_generic_rw_locked(d, state->buf, sizeof(state->buf),
+	ret = mediatree_dvb_usbv2_generic_rw_locked(d, state->buf, sizeof(state->buf),
 			state->buf, sizeof(state->buf));
 	if (ret)
 		goto error_unlock;
@@ -1398,11 +1398,11 @@ MODULE_DEVICE_TABLE(usb, anysee_id_table);
 static struct usb_driver anysee_usb_driver = {
 	.name = KBUILD_MODNAME,
 	.id_table = anysee_id_table,
-	.probe = dvb_usbv2_probe,
-	.disconnect = dvb_usbv2_disconnect,
-	.suspend = dvb_usbv2_suspend,
-	.resume = dvb_usbv2_resume,
-	.reset_resume = dvb_usbv2_reset_resume,
+	.probe = mediatree_dvb_usbv2_probe,
+	.disconnect = mediatree_dvb_usbv2_disconnect,
+	.suspend = mediatree_dvb_usbv2_suspend,
+	.resume = mediatree_dvb_usbv2_resume,
+	.reset_resume = mediatree_dvb_usbv2_reset_resume,
 	.no_dynamic_id = 1,
 	.soft_unbind = 1,
 };
